Content.Server/Kitchen/EntitySystems/DeepFryerSystem.cs

```diff
@@ -0,0 +1,1004 @@
+        }
+
+        private void UpdateAmbientSound(EntityUid uid, DeepFryerComponent component)
+        {
+            if (TryComp<AmbientSoundComponent>(uid, out var ambientSoundComponent))
+            {
+                bool enabled = HasBubblingOil(uid, component);
+                if (enabled != ambientSoundComponent.Enabled)
+                {
+                    ambientSoundComponent.Enabled = enabled;
+                    Dirty(ambientSoundComponent);
+                }
+            }
+        }
+
+        private void UpdateNextFryTime(EntityUid uid, DeepFryerComponent component)
+        {
+            component.NextFryTime = _gameTimingSystem.CurTime + component.FryInterval;
+        }
+
+        /// <summary>
+        /// Make an item look deep-fried.
+        /// </summary>
+        private void MakeCrispy(EntityUid item)
+        {
+            EnsureComp<ServerAppearanceComponent>(item);
+            EnsureComp<DeepFriedComponent>(item);
+
+            _appearanceSystem.SetData(item, DeepFriedVisuals.Fried, true);
+        }
+
+        /// <summary>
+        /// Turn a dead mob into food.
+        /// </summary>
+        /// <remarks>
+        /// This is meant to be an irreversible process, similar to gibbing.
+        /// </remarks>
+        public bool TryMakeMobIntoFood(EntityUid mob, MobStateComponent mobStateComponent, bool force = false)
+        {
+            // Don't do anything to mobs until they're dead.
+            if (force || _mobStateSystem.IsDead(mob, mobStateComponent))
+            {
+                RemComp<ActiveNPCComponent>(mob);
+                RemComp<AtmosExposedComponent>(mob);
+                RemComp<BarotraumaComponent>(mob);
+                RemComp<BuckleComponent>(mob);
+                RemComp<GhostTakeoverAvailableComponent>(mob);
+                RemComp<InternalsComponent>(mob);
+                RemComp<PerishableComponent>(mob);
+                RemComp<RespiratorComponent>(mob);
+                RemComp<RottingComponent>(mob);
+
+                // Ensure it's Food here, so it passes the whitelist.
+                var mobFoodComponent = EnsureComp<FoodComponent>(mob);
+                var mobFoodSolution = _solutionContainerSystem.EnsureSolution(mob, mobFoodComponent.SolutionName, out bool alreadyHadFood);
+
+                // This line here is mainly for mice, because they have a food
+                // component that mirrors how much blood they have, which is
+                // used for the reagent grinder.
+                if (alreadyHadFood)
+                    _solutionContainerSystem.RemoveAllSolution(mob, mobFoodSolution);
+
+                if (TryComp<BloodstreamComponent>(mob, out var bloodstreamComponent))
+                {
+                    // Fry off any blood into protein.
+                    var bloodSolution = bloodstreamComponent.BloodSolution;
+                    var removedBloodQuantity = bloodSolution.RemoveReagent("Blood", FixedPoint2.MaxValue);
```

Comment:
It's only meant to turn regular blood into protein. I'm not yet handling any non-blood blood. It should do nothing in the event that there's no actual blood, but I could put in an if statement to prevent any further processing.